### PR TITLE
Add new items from the Ohai release notes

### PR DESCRIPTION
**Pull request by tas50**
Add 2 more breaking changes for Chef 15.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -111,6 +111,14 @@ node.default["foo"][0][:bar]  # does not work due to the sub-Hash not
 
 The new behavior uses a Mash so that the attributes will work as expected.
 
+### Ohai's system_profile plugin for macOS removed
+
+The system_profile plugin no longer ships in Ohai. This plugin did not correctly return data on modern Mac systems. Additionally the same data is provided by the hardware plugin, which has a format that is simpler to consume. Removing this plugin speeds up Ohai (and Chef) by ~3 seconds and greatly reduces the size of the node object on the Chef server.
```
[the remark this answers is not in the document]
Comment (tas50) [reply RELEASE_NOTES.md]: That first sentence feels pretty run on ish.

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -111,6 +111,14 @@ node.default["foo"][0][:bar]  # does not work due to the sub-Hash not
...
+
+### Ohai's Ohai::Util::Win32::GroupHelper class has been removed
+
+The Ohai::Util::Win32::GroupHelper helper class has been removed from Ohai. This class was only meant to be used internally within several Windows plugins, but it was never marked private in the codebase. If any Ohai plugins rely on this helper class, you will need to update your plugins for Ohai 15.
```
[the remark this answers is not in the document]
Comment (tas50) [reply RELEASE_NOTES.md]: Updated

File: RELEASE_NOTES.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tas50) [reply RELEASE_NOTES.md]: Let me know about the update I pushed @kagarmoe

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -111,6 +111,14 @@ node.default["foo"][0][:bar]  # does not work due to the sub-Hash not
 
 The new behavior uses a Mash so that the attributes will work as expected.
 
+### Ohai's system_profile plugin for macOS removed
+
+We removed the system_profile plugin because it incorrectly returned data on modern Mac systems. If you relied on this plugin you'll want to update recipes to use `node['hardware']` instead, which correctly returns the same data, but in a more easily consumed format. Removing this plugin speeds up Ohai (and Chef) by ~3 seconds and dramatically reduces the size of the node object on the Chef server.
```
[the remark this answers is not in the document]
Comment (kagarmoe) [reply RELEASE_NOTES.md]: I like Mary's change :-)

**Pull request merged**